sdk/core/AzureCore/Source/Pipeline/Transport/URLSessionTransport.swift

```diff
@@ -103,11 +103,19 @@ public class URLSessionTransport: TransportStage {
                     return
                 }
             }
-
             let rawResponse = response as? HTTPURLResponse
             let httpResponse = URLHTTPResponse(request: httpRequest, response: rawResponse)
             httpResponse.data = data
 
+            // check for invalid status codes
+            let statusCode = httpResponse.statusCode ?? -1
+            let allowedStatusCodes = responseContext?.value(forKey: .allowedStatusCodes) as? [Int] ?? [200]
```

Comment:
Should we change the default to permit any 2xx response code?

Reply:
We should consult with other languages and see what they do. Either way, I would recommend doing that as a separate PR, if necessary.

Reply:
Sounds good to me